instrumentation/opentelemetry-instrumentation-confluent-kafka/src/opentelemetry/instrumentation/confluent_kafka/__init__.py

```diff
@@ -361,3 +377,35 @@ def wrap_poll(func, instance, tracer, args, kwargs):
         )
 
         return record
+
+    @staticmethod
+    def wrap_consume(func, instance, tracer, args, kwargs):
+        if instance._current_consume_span:
+            context.detach(instance._current_context_token)
+            instance._current_context_token = None
+            instance._current_consume_span.end()
+            instance._current_consume_span = None
+
+        with tracer.start_as_current_span(
+            "recv", end_on_exit=True, kind=trace.SpanKind.CONSUMER
+        ):
+            records = func(*args, **kwargs)
+            if len(records) > 0:
+                links = _get_links_from_records(records)
+                instance._current_consume_span = tracer.start_span(
+                    name=f"{records[0].topic()} process",
+                    links=links,
+                    kind=SpanKind.CONSUMER,
+                )
+
+                _enrich_span(
```

Comment:
I'm out of my depth here, but I'm noticing that this method is very similar to the one above it, which passes in the record partition and offset to `_enrich_span()`. Do we perhaps want to do the same here (and if so, should duplicated code be factored out)?

Reply:
The `partition` and `offset` arguments are used to compute the `messaging.message.id` attribute in the `_enrich_span` function, here is the piece of code:

```python
    if partition is not None and offset is not None and topic:
        span.set_attribute(
            SpanAttributes.MESSAGING_MESSAGE_ID,
            f"{topic}.{partition}.{offset}",
        )
```

In the `poll` method, you process one record at a time so it's easy to identify the `messaging.message.id`. The problem with the `consume` method is that you can get multiple records belonging to different partitions at different offsets but the corresponding span can only contain one `messaging.message.id`. In fact, the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#batch-processing) leaves the `messaging.message.id` empty for batch processing spans.

However, I agree with you, both functions `wrap_poll` and `wrap_consume` look similar. I factored out the similarities in two new functions: `_end_current_consume_span` and `_create_new_consume_span`. Let me know what you think! 😄